subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -660,36 +718,60 @@ static void dhcpv4_enter_requesting(struct net_if *iface, struct dhcp_msg *msg)
 /* Must be invoked with lock held */
 static void dhcpv4_enter_bound(struct net_if *iface)
 {
-	uint32_t renewal_time;
-	uint32_t rebinding_time;
+	struct net_if_dhcpv4 *dhcpv4 = &iface->config.dhcpv4;
 
-	renewal_time = iface->config.dhcpv4.renewal_time;
-	if (!renewal_time) {
+	/* Load defaults in case server did not provide T1/T2 values. */
+	if (dhcpv4->renewal_time == 0) {
 		/* The default renewal time rfc2131 4.4.5 */
-		renewal_time = iface->config.dhcpv4.lease_time / 2U;
-		iface->config.dhcpv4.renewal_time = renewal_time;
+		dhcpv4->renewal_time = dhcpv4->lease_time / 2U;
 	}
 
-	rebinding_time = iface->config.dhcpv4.rebinding_time;
-	if (!rebinding_time) {
+	if (dhcpv4->rebinding_time == 0) {
 		/* The default rebinding time rfc2131 4.4.5 */
-		rebinding_time = iface->config.dhcpv4.lease_time * 875U / 1000;
-		iface->config.dhcpv4.rebinding_time = rebinding_time;
+		dhcpv4->rebinding_time = dhcpv4->lease_time * 875U / 1000;
```

Comment:
Makes probably no difference but we should ise 1000U here, also noticed similar things in other functions

Reply:
I'll look over the diff and fix where applicable.

Reply:
Should be fixed now.